# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/GCMarker.h
# Commit: 7cda151d97a5
# Full Hash: 7cda151d97a53e47a1f526390dbb005ba8f78fca
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-10-28 21:22:11
# Regressor Bug: 1797755
# File Overlap Count: 1
# Description:
#   Bug 1797755 - Part 5: Use a single initial mark stack size regardless of whether incremental GC is enabled r=sfink
#   
#   Currently we initialize the mark stack to a different size depending on whether
#   or not incremental GC is enable. However, after the first GC we always shrink
#   it to the initial size when it is disabled.
# ==============================================================================

diff -r 098569654f17 -r 7cda151d97a5 js/src/gc/GCMarker.h
--- a/js/src/gc/GCMarker.h	Fri Oct 28 09:47:14 2022 +0000
+++ b/js/src/gc/GCMarker.h	Fri Oct 28 09:47:15 2022 +0000
@@ -21,8 +21,7 @@
 class SliceBudget;
 class WeakMapBase;
 
-static const size_t NON_INCREMENTAL_MARK_STACK_BASE_CAPACITY = 4096;
-static const size_t INCREMENTAL_MARK_STACK_BASE_CAPACITY = 32768;
+static const size_t MARK_STACK_BASE_CAPACITY = 4096;
 
 enum class SlotsOrElementsKind { Elements, FixedSlots, DynamicSlots };
 
@@ -138,8 +137,8 @@
 
   size_t position() const { return topIndex_; }
 
-  [[nodiscard]] bool init(bool incrementalGCEnabled);
-  [[nodiscard]] bool setStackCapacity(bool incrementalGCEnabled);
+  [[nodiscard]] bool init();
+  [[nodiscard]] bool resetStackCapacity();
 
 #ifdef JS_GC_ZEAL
   void setMaxCapacity(size_t maxCapacity);
@@ -162,13 +161,7 @@
   TaggedPtr popPtr();
   SlotsOrElementsRange popSlotsOrElementsRange();
 
-  void clear() {
-    // Fall back to the smaller initial capacity so we don't hold on to excess
-    // memory between GCs.
-    stack().clearAndFree();
-    (void)stack().resize(NON_INCREMENTAL_MARK_STACK_BASE_CAPACITY);
-    topIndex_ = 0;
-  }
+  void clear();
 
   void poisonUnused();
 
@@ -325,11 +318,6 @@
   [[nodiscard]] bool markUntilBudgetExhausted(
       SliceBudget& budget, ShouldReportMarkTime reportTime = ReportMarkTime);
 
-  void setIncrementalGCEnabled(bool enabled) {
-    // Ignore failure to resize the stack and keep using the existing stack.
-    (void)stack.setStackCapacity(enabled);
-  }
-
   size_t sizeOfExcludingThis(mozilla::MallocSizeOf mallocSizeOf) const;
 
 #ifdef DEBUG